### PR TITLE
feat: add tooling for airlock

### PR DESCRIPTION
**Pull request by madwort**
* we now have the required image available
* no CI tests yet, but this is running on the test backend

### CODE REVIEW

File: tests/check-airlock.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+set -euxo pipefail
+
+# set DNS for RELEASE_HOST to 127.0.0.1
+# shellcheck source=/dev/null
+source /home/opensafely/config/load-env
+HOSTNAME="$(echo "$RELEASE_HOST" | cut -d'/' -f3 | cut -d':' -f1)"
+grep -q "$HOSTNAME" /etc/hosts || echo "127.0.0.1 $HOSTNAME" >> /etc/hosts
+
+# Do a basic smoke test
+curl -I "https://localhost:8000" -s --compressed --fail --retry 20 --retry-delay 1
```
Comment (madwort) [inline tests/check-airlock.sh:11]: have tried `localhost:8000` as well as `$RELEASE_HOST`

File: services/airlock/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+the :latest tag.
+
+The container is always named `airlock`, and can be managed using
+normal docker tools or via `docker compose`.
+
+# Configuration
+
+airlock shares the same configuration as jobrunner, which currently lives
+in `/home/opensafely/config/*.env`. In particular, it uses `AIRLOCK_API_TOKEN` to
+sign and validate requests. This currently duplicates `JOB_SERVER_TOKEN`.
```
Comment (bloodearnest) [inline services/airlock/README.md:14]: Yeah, I think we should rename the env var in airlock to JOB_SERVER_TOKEN, as its shared between apps and pre-exists.

File: services/airlock/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+The default install will generate self-signed certifcates, but these can replaced.
+
+# Useful commands
+
+All these commands assume you are runing from `~opensafely/airlock` directory.
+
+ - *restart*: docker compose restart
+ - *logs*: docker compose logs
+ - *root shell*: docker-compose exec -u 0 airlock bash
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply services/airlock/README.md:28]: If we're shutting down EMIS, we might be able use v2, as its available on TPP and test backend.

File: services/airlock/install.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+fi
+
+if test "${TEST:-}" = "true"; then
+    just -f $DIR/justfile create-test-certificates
+fi
+
+chown -R opensafely:opensafely "$DIR"
+chmod -R go-rwx "$DIR"
+
+RELEASES_DIR="$MEDIUM_PRIVACY_STORAGE_BASE/releases"
```
Comment (bloodearnest) [inline services/airlock/install.sh:22]:
```suggestion
REQUESTS_DIR="$MEDIUM_PRIVACY_STORAGE_BASE/requests"
```

File: services/airlock/install.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+mkdir -p "$RELEASES_DIR"
```
Comment (bloodearnest) [inline services/airlock/install.sh:23]:
```suggestion
mkdir -p "$REQUESTS_DIR"
```

File: services/airlock/install.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+find "$RELEASES_DIR" -type f -exec chmod 640 {} +
```
Comment (bloodearnest) [inline services/airlock/install.sh:24]:
```suggestion
find "$REQUESTS_DIR" -type f -exec chmod 640 {} +
```

**Review by bloodearnest — APPROVED** [review]
Looks good, thanks

Couple of minor suggestions to consistently use REQUESTS rather than RELEASES nameing

File: services/airlock/README.md (hunk 2 above)
Comment (madwort) [reply services/airlock/README.md:14]: agree, I will open a ticket on the airlock repo - what should we do here & now, though? do you want to do something like 

```bash
AIRLOCK_API_TOKEN="$JOB_SERVER_TOKEN"
```

or merge as-is for now?

Comment (madwort) [reply services/airlock/README.md:14]: gonna merge as-is

**Pull request merged**